### PR TITLE
[FLINK-36115][pipeline-connector][mysql] add scan.incremental.newly-added-table.enabled option

### PR DESCRIPTION
**Pull request by lvyanquan**
This allow to scan newly table DDL during incremental reading stage.

### CODE REVIEW

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java [hunk 1]
```diff
@@ -261,4 +261,12 @@ public class MySqlDataSourceOptions {
                                     + "If there is a need to use a dot (.) in a regular expression to match any character, "
                                     + "it is necessary to escape the dot with a backslash."
                                     + "eg. db0.\\.*, db1.user_table_[0-9]+, db[1-2].[app|web]_order_\\.*");
+
+    @Experimental
+    public static final ConfigOption<Boolean> SCAN_INCREMENTAL_NEWLY_ADDED_TABLE_ENABLED =
+            ConfigOptions.key("scan.incremental.newly-added-table.enabled")
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java:267]: scan.binlog.newly-added-table.enabled

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java [hunk 2]
```diff
@@ -261,4 +261,12 @@ public class MySqlDataSourceOptions {
...
+                    .booleanType()
+                    .defaultValue(false)
+                    .withDescription(
+                            "Whether to scan the ddl and dml statements of newly added tables or not in incremental reading stage, by default is false.");
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/MySqlDataSourceOptions.java:271]: We should explain the option works phase and the behavior, and also the diff with SCAN_NEWLY_ADDED_TABLE_ENABLED  : do re-snapshot & binlog-reading for newly added table when restore
SCAN_BINLOG_NEWLY_ADDED_TABLE_ENABLED ： only do binlog-reading for newly added table  during binlog reading phase

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableITCase.java [hunk 3]
```diff
@@ -149,6 +153,40 @@ private MySqlConnection getConnection() {
...
+                        new EventTypeInfo());
+
+        TypeSerializer<Event> serializer =
+                source.getTransformation().getOutputType().createSerializer(env.getConfig());
+        CheckpointedCollectResultBuffer<Event> resultBuffer =
+                new CheckpointedCollectResultBuffer<>(serializer);
+        String accumulatorName = "dataStreamCollect_" + UUID.randomUUID();
+        CollectResultIterator<Event> iterator =
+                addCollector(env, source, resultBuffer, serializer, accumulatorName);
+        env.executeAsync("AddNewlyTablesInIncrementalStage");
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/test/java/org/apache/flink/cdc/connectors/mysql/source/MysqlPipelineNewlyAddedTableITCase.java:183]: binlogReadingState

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java [hunk 4]
```diff
@@ -128,6 +131,8 @@ public DataSource createDataSource(Context context) {
         int connectMaxRetries = config.get(CONNECT_MAX_RETRIES);
         int connectionPoolSize = config.get(CONNECTION_POOL_SIZE);
         boolean scanNewlyAddedTableEnabled = config.get(SCAN_NEWLY_ADDED_TABLE_ENABLED);
+        boolean scanIncrementalNewlyAddedTableEnabled =
```
Comment (leonardBang) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java:134]:
```suggestion
        boolean scanInBinlogPhaseNewlyAddedTableEnabled =
```

**Review by Jiabao-Sun — APPROVED** [review]
LGTM

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java (hunk 4 above)
Comment (zhongyuandong) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/factory/MySqlDataSourceFactory.java]: Which version does it support? How to set it up?

**Pull request merged**